CONTRIBUTING.md

```diff
@@ -42,31 +42,89 @@ recommended for production deployments). Determined has been tested with Ubuntu
 recommended; on AWS, a good AMI to use is a recent version of "Deep Learning
 Base AMI (Ubuntu)".
 
-Start by cloning the Determined repo:
+Start by cloning the Determined repository:
 
 ```sh
-git clone --recurse-submodules git@github.com:determined-ai/determined.git
+git clone --recurse-submodules https://github.com/determined-ai/determined.git
 ```
 
 #### Prerequisites
 
 - Go (>= 1.18)
-- Python (>= 3.6, <= 3.9)
+- Python (>= 3.6, <= 3.9), including:
+   - python3-venv
+   - python3-wheel
+   - python3-dev
 - Node (>= 16.13, < 17)
 - NPM (>= 8)
 - Docker (>= 19.03)
 - Helm (>= 3.0.0)
-- Protoc (>= 3.0)
+- protobuf-compiler (>= 3.0)
 - Java (>= 7)
 - cURL (>= 7)
 - jq (>= 1.6)
 - socat (>= 1.7)
 
+If you are installing prerequisites from your Linux distribution's package 
+repository, ensure that they meet the version requirements above, particularly 
+Python and Node.
+
+#### Install Prerequisites with Homebrew
+
+Because the versions of prerequisites from Linux distribution package 
+repositories can vary widely, we recommend installing the Determined build 
+prerequisites with [Homebrew](https://brew.sh/).
+
+The following guide is also applicable for building Determined on macOS.
+
+Install a compiler and build tools:
+
+- macOS: Install the [Command Line Tools for Xcode](https://developer.apple.com/) from Apple
+- Debian and Ubuntu: `sudo apt install build-essential`
+- Red Hat, CentOS, and Fedora: `sudo yum install gcc make perl-devel`
+- SUSE and openSUSE: `sudo zypper install -t pattern devel_basis`
+
+Install Homebrew:
+
+```sh
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+Add Homebrew to your PATH:
+
+```sh
+echo 'eval "$($HOME/.linuxbrew/bin/brew shellenv)"' >> $HOME/.profile
+eval "$($HOME/.linuxbrew/bin/brew shellenv)"
+```
+
+Install the Determined prerequisites:
+
+```sh
+brew install go@1.18 python@3.7 node@16 openjdk@11 protobuf docker helm curl jq socat
+```
+
+Add Python, Node, and openJDK to your PATH:
+
+```sh
+echo 'export PATH="$HOME/.linuxbrew/opt/python@3.7/bin:$HOME/.linuxbrew/opt/node@16/bin:$HOME/.linuxbrew/opt/openjdk@11/bin:$PATH"' >> $HOME/.profile
+source $HOME/.profile
+```
+
+On Red Hat and CentOS 8 and Ubuntu 16.04, add the compiled GCC 11 libraries to your LD_LIBRARY_PATH:
```

Comment:
It is old. But there are still a lot of Ubuntu 16.04 users out there.

The version number is significant because more recent LTS releases of Ubuntu (18.04,20.04,22.04) have recent enough glibc on their systems to run Node 16.

But Ubuntu 16.04 and RHEL 8 need to be pointed at the glibc brought down by Homebrew with GCC11.